src/SprykerMiddleware/Zed/Process/Business/Stream/JsonStream.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace SprykerMiddleware\Zed\Process\Business\Stream;
+
+// @codingStandardsIgnoreFile
+class JsonStream
+{
+    /**
+     * @var resource
+     */
+    protected $subhandle;
+
+    /**
+     * @var array
+     */
+    protected $data = [];
+
+    /**
+     * @var string
+     */
+    protected $mode = '';
+
+    /**
+     * @var int
+     */
+    protected $position;
+
+    /**
+     * @return bool
+     */
+    public function stream_close()
+    {
+        return fclose($this->subhandle);
+    }
+
+    /**
+     * @return bool
+     */
+    public function stream_eof()
+    {
+        return $this->position >= count($this->data);
+    }
+
+    /**
+     * @return bool
+     */
+    public function stream_flush()
+    {
+        fwrite($this->subhandle, json_encode($this->data));
+        return fflush($this->subhandle);
+    }
+
+    /**
+     * @param string $data
+     *
+     * @return void
```

Comment:
Should return the number of bytes that were successfully stored, or 0 if none could be stored.